### PR TITLE
Convert to vue.js

### PR DESCRIPTION
**Pull request by donggyu04**
Convert freemarker to vue.js

 - Use npm for package manager.
 - Use webpack for module bundler.
 - Use vue.js framework for single page application.

### CODE REVIEW

File: ngrinder-controller/src/main/java/org/ngrinder/perftest/controller/PerfTestController.java [hunk 1]
```diff
@@ -785,8 +744,8 @@ public boolean apply(@Nullable FileEntry input) {
 	 * @return json message containing test info.
 	 */
 	@RestAPI
-	@RequestMapping(value = "/api/{id}", method = RequestMethod.GET)
-	public HttpEntity<String> getOne(User user, @PathVariable("id") Long id) {
+	@GetMapping(value = "/api/{id}")
+	public HttpEntity<String> getOne(User user, @PathVariable Long id) {
```
Comment (songeunwoo) [inline ngrinder-controller/src/main/java/org/ngrinder/perftest/controller/PerfTestController.java]: That method should be moved to ```PerfTestApiController```

Comment (donggyu04) [reply ngrinder-controller/src/main/java/org/ngrinder/perftest/controller/PerfTestController.java]: Yes, currently I moved only modified method to PerftestApiController.
this seems like i missed. 

I will clean up all methods at the end of the perftest related implementation.

Comment (songeunwoo) [reply ngrinder-controller/src/main/java/org/ngrinder/perftest/controller/PerfTestController.java]: It's great.

File: ngrinder-controller/src/main/java/org/ngrinder/script/controller/FileEntryApiController.java [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+	/**
+	 * Search files on the query.
+	 *
+	 * @param user  current user
+	 * @param query query string
+	 * @return list of filtered files
+	 */
+	@RestAPI
+	@GetMapping("/search")
+	public HttpEntity<String> search(User user, @RequestParam(required = true, value = "query") final String query) {
```
Comment (songeunwoo) [inline ngrinder-controller/src/main/java/org/ngrinder/script/controller/FileEntryApiController.java]: required is true by default.
Please delete it.

File: ngrinder-controller/src/main/java/org/ngrinder/script/controller/FileEntryApiController.java [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+	 * @param path                  path in which a file will be added
+	 * @param testUrl               url which the script may use
+	 * @param fileName              fileName
+	 * @param scriptType            Type of script. optional
+	 * @param createLibAndResources true if libs and resources should be created as well.
+	 * @return response map
+	 */
+	@PostMapping(value = "/new/**", params = "type=script")
+	public Map<String, Object> createForm(User user, @RemainedPath String path,
+	                                      @RequestParam(value = "testUrl", required = false) String testUrl,
```
Comment (songeunwoo) [inline ngrinder-controller/src/main/java/org/ngrinder/script/controller/FileEntryApiController.java]: What if we set the default url to defaultValue?
```
@RequestParam(value = "testUrl", required = false) String testUrl,
=>
@RequestParam(value = "testUrl", required = false, defaultValue = "http://please_modify_this.com") String testUrl,
```

File: ngrinder-controller/src/main/java/org/ngrinder/script/controller/FileEntryApiController.java [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+	                                      @RequestParam("fileName") String fileName,
+	                                      @RequestParam(value = "scriptType", required = false) String scriptType,
+	                                      @RequestParam(value = "createLibAndResource", defaultValue = "false") boolean createLibAndResources,
+	                                      @RequestParam(value = "options", required = false) String options) {
```
Comment (songeunwoo) [inline ngrinder-controller/src/main/java/org/ngrinder/script/controller/FileEntryApiController.java]: Please check the code convention.
I do not think this is a tab around here.

File: ngrinder-controller/src/main/java/org/ngrinder/script/controller/FileEntryApiController.java [hunk 5]
```diff
@@ -0,0 +1,386 @@
...
+		return successJsonHttpEntity();
+	}
+
+	/**
+	 * Get the script path BreadCrumbs HTML string.
+	 *
+	 * @param path path
+	 * @return generated HTML
+	 */
+	public String getScriptPathBreadcrumbs(String path) {
```
Comment (songeunwoo) [inline ngrinder-controller/src/main/java/org/ngrinder/script/controller/FileEntryApiController.java]: I think the method here is also private.

File: ngrinder-controller/src/main/java/org/ngrinder/script/controller/FileEntryApiController.java [hunk 6]
```diff
@@ -0,0 +1,386 @@
...
+	/**
+	 * Delete files on the given path.
+	 *
+	 * @param user        user
+	 * @param path        base path
+	 * @param filesString file list delimited by ","
+	 * @return json string
+	 */
+	@PostMapping(value = "/delete/**")
+	@ResponseBody
```
Comment (songeunwoo) [inline ngrinder-controller/src/main/java/org/ngrinder/script/controller/FileEntryApiController.java]: Please do not need ResponseBody.

File: ngrinder-controller/src/main/java/org/ngrinder/script/controller/FileEntryApiController.java [hunk 7]
```diff
@@ -0,0 +1,386 @@
...
+	 * @param user        user
+	 * @param path        path
+	 * @param description description
+	 * @param file        multi part file
+	 * @return success json string
+	 */
+	@RestAPI
+	@PostMapping(value = "/**", params = "action=upload")
+	public HttpEntity<String> uploadForAPI(User user, @RemainedPath String path,
+										   @RequestParam("description") String description,
```
Comment (songeunwoo) [inline ngrinder-controller/src/main/java/org/ngrinder/script/controller/FileEntryApiController.java]: Please check the code convention.

**Comment by junoyoon** [comment]
LGTM.

**Pull request merged**